wrangler-core/src/main/java/io/cdap/wrangler/executor/RecipePipelineExecutor.java

```diff
@@ -122,14 +137,26 @@ public List<Row> execute(List<Row> rows) throws RecipeException {
               if (cumulativeRows.size() < 1) {
                 break;
               }
+              if (designTime && schema != null) {
+                Schema directiveOutputSchema = directive.getOutputSchema(schema);
+                schema = directiveOutputSchema != null ? directiveOutputSchema
+                  : generateOutputSchema(schema, cumulativeRows);
```

Comment:
We should introduce a feature flag for this with default enabled.. If schema generator fails on some customer schema, there should be a way to disable it